model/flow/identifier.go

```diff
@@ -121,6 +121,18 @@ func MerkleRoot(ids ...Identifier) Identifier {
 	return root
 }
 
+func MerkleRootOfList(entities []IDWithFingerprint) Identifier {
+	var root Identifier
+	tree := merkle.NewTree()
+	for _, e := range entities {
+		id := e.ID()
+		tree.Put(id[:], e.Fingerprint())
```

Comment:
Btw, can't the key of the tree be the index of the event in the input list? This avoids the extra hashing to compute the ID and allows the tree to remain shallow.

Reply:
`EventID` is already used in many places, including SDK so I don't want to change semantics.
But now I think we can kill two birds with one stone - have separate tree path which can be used as tree key here, but it will require use to keep the new interface, renaming it a bit

Reply:
Switched to hashing list as discussed, so I it doesn't apply anymore